src/Gir/SymbolTable.cs

```diff
@@ -0,0 +1,310 @@
+			AddType(new Primitive("unsigned int", "uint", "0"));
+			AddType(new Primitive("unsigned-int", "uint", "0"));
+			AddType(new Primitive("gshort", "short", "0"));
+			AddType(new Primitive("gushort", "ushort", "0"));
+			AddType(new Primitive("short", "short", "0"));
+			AddType(new Primitive("guchar", "byte", "0"));
+			AddType(new Primitive("unsigned char", "byte", "0"));
+			AddType(new Primitive("unsigned-char", "byte", "0"));
+			AddType(new Primitive("guint1", "bool", "false"));
+			AddType(new Primitive("uint1", "bool", "false"));
+			AddType(new Primitive("gint8", "sbyte", "0"));
+			AddType(new Primitive("guint8", "byte", "0"));
+			AddType(new Primitive("gint16", "short", "0"));
+			AddType(new Primitive("guint16", "ushort", "0"));
+			AddType(new Primitive("gint32", "int", "0"));
+			AddType(new Primitive("guint32", "uint", "0"));
+			AddType(new Primitive("gint64", "long", "0"));
+			AddType(new Primitive("guint64", "ulong", "0"));
+			AddType(new Primitive("long long", "long", "0"));
+			AddType(new Primitive("gfloat", "float", "0.0"));
+			AddType(new Primitive("float", "float", "0.0"));
+			AddType(new Primitive("gdouble", "double", "0.0"));
+			AddType(new Primitive("double", "double", "0.0"));
+			AddType(new Primitive("goffset", "long", "0"));
+			AddType(new Primitive("GQuark", "int", "0"));
+
+			// platform specific integer types.
+#if WIN64LONGS
+			AddType (new Primitive ("long", "int", "0"));
+			AddType (new Primitive ("glong", "int", "0"));
+			AddType (new Primitive ("ulong", "uint", "0"));
+			AddType (new Primitive ("gulong", "uint", "0"));
+			AddType (new Primitive ("unsigned long", "uint", "0"));
+#else
+			AddType(new IntPtrGen("long"));
+			AddType(new IntPtrGen("glong"));
+			AddType(new UIntPtrGen("ulong"));
+			AddType(new UIntPtrGen("gulong"));
+			AddType(new UIntPtrGen("unsigned long"));
+#endif
+
+			AddType(new IntPtrGen("ssize_t"));
+			AddType(new IntPtrGen("gssize"));
+			AddType(new UIntPtrGen("size_t"));
+			AddType(new UIntPtrGen("gsize"));
+
+#if OFF_T_8
+			AddType (new AliasGen ("off_t", "long"));
+#else
+			AddType(new IntPtrGen("off_t"));
+#endif
+
+			// string types
+			//AddType(new ConstStringGen("const-gchar"));
+			//AddType(new ConstStringGen("const-xmlChar"));
+			//AddType(new ConstStringGen("const-char"));
+			//AddType(new ConstFilenameGen("const-gfilename"));
+			//AddType(new StringMarshalGen("gfilename", "string", "IntPtr", "GLib.Marshaller.StringToFilenamePtr({0})", "GLib.Marshaller.FilenamePtrToStringGFree({0})"));
+			//AddType(new StringMarshalGen("gchar", "string", "IntPtr", "GLib.Marshaller.StringToPtrGStrdup({0})", "GLib.Marshaller.PtrToStringGFree({0})"));
+			//AddType(new StringMarshalGen("char", "string", "IntPtr", "GLib.Marshaller.StringToPtrGStrdup({0})", "GLib.Marshaller.PtrToStringGFree({0})"));
+			AddType(new Primitive("GStrv", "string[]", "null"));
+
+			// manually wrapped types requiring more complex marshaling
```

Comment:
I don't think we want any of these from here on. Some are not bound correctly this way and we can generate these from GLib.gir.

Reply:
IF we can pick these up from the gir files I think that is a much better idea.

Reply:
All of them are in the gir files. We can even generate the whole GLib if the generator is complete and works. We can then add .NET integration with these classes on top of the generated classes.